oxidation/libparsec/crates/types/src/addr.rs

```diff
@@ -99,11 +94,47 @@ macro_rules! impl_common_stuff {
             type Err = &'static str;
 
             fn from_str(url: &str) -> Result<Self, Self::Err> {
-                // Note `Url::parse` takes care of percent-encoding for query params
-                let parsed = Url::parse(url).map_err(|_| "Invalid URL")?;
-                let pairs = parsed.query_pairs();
+                lazy_static! {
+                    static ref SCHEME_WITH_SLASHES: String = format!("{}://", PARSEC_SCHEME);
+                }
+                lazy_static! {
+                    static ref SCHEME_ERROR_MSG: String =
+                        format!("Must start with `{}://`", PARSEC_SCHEME);
+                }
```

Comment:
Use constant value instead (use `concat!`)

https://doc.rust-lang.org/std/macro.concat.html

Reply:
This doesn't work because concat only allow literal (so cannot use `PARSEC_SCHEME`)